iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/sync/IoTDBThriftSyncConnector.java

```diff
@@ -262,8 +262,9 @@ public void transfer(TabletInsertionEvent tabletInsertionEvent) throws Exception
 
     try {
       if (isTabletBatchModeEnabled) {
-        if (tabletBatchBuilder.onEvent(tabletInsertionEvent)) {
-          doTransfer(client);
+        if (tabletBatchBuilder.onEvent(
+            tabletInsertionEvent, ((EnrichedEvent) tabletInsertionEvent).getCommitId())) {
```

Comment:
If the commitId is built in the event itself, maybe it's adequate to pass only the event in.